File: jesterclub/jcangularwebclient/src/app/models/joke.ts
import { EmotionCounter } from './emotioncounter';
import { ResponseStatistic } from './responsestatistic';

export interface Joke {
  Id: string;

  UserEmail: string;

  UserName: string;

  Text: string;

  Source: string;

  Copyright: string;

  CreationDate: Date;

  ReleasedDate?: Date;

  Tags: Array<string>;

  EmotionCounters: Array<EmotionCounter>;

  ResponseSum: number;

  ResponseStatistics: Array<ResponseStatistic>;
}

File: jesterclub/jcsqlserverservice/Models/Joke.cs
using System;
using System.Collections.Generic;

namespace jcsqlserverservice.Models {
    public class Joke {

        public int? JokeId { get; set; }

        public string JokeText { get; set; }

        public string Source { get; set; }

        public string Copyright { get; set; }

        public DateTime CreationDate { get; set; }

        public DateTime? ReleasedDate { get; set; }

        public int ResponseSum { get; set; }

        public User User { get; set; }

        public ICollection<Tag> Tags { get; set; }

        public ICollection<EmotionCounter> EmotionCounters { get; set; }

        public ICollection<ResponseStatistic> ResponseStatistics { get; set; }
    }
}